lowranceusr.cc

```diff
@@ -423,28 +975,399 @@ lowranceusr_parse_waypt(Waypoint* wpt_tmp)
 
   /* Waypoint Type (USER, TEMPORARY, POINT_OF_INTEREST) */
   short waypt_type = gbfgetint16(file_in);
-  if (global_opts.debug_level >= 1) {
-    printf(MYNAME " parse_waypt: waypt_type = %d\n",waypt_type);
+  if (global_opts.debug_level > 2) {
+    if (global_opts.debug_level == 99) {
+      printf(" %04x (%d)", (int)waypt_type, (int)waypt_type);
+    } else {
+      printf(MYNAME " parse_waypt: waypt_type = %d\n",waypt_type);
+    }
   }
 
   // Version 3 has a depth field here.
-  if (reading_version >= 3) {
+  if (reading_version == 3) {
     float depth_feet = gbfgetflt(file_in);
     if (std::abs(depth_feet - 99999.0)  > .1) {
       WAYPT_SET(wpt_tmp, depth, FEET_TO_METERS(depth_feet));
+      if (global_opts.debug_level == 99) {
+        printf("   %10.1f", depth_feet);
+      }
+    } else {
+      if (global_opts.debug_level == 99) {
+        printf("      UNKNOWN");
+      }
     }
   }
 
+  if (global_opts.debug_level == 99) {
+    printf("\n");
+  }
 }
 
+static void
+lowranceusr4_parse_waypt(Waypoint* wpt_tmp)
+{
+  char name_buff[MAXUSRSTRINGSIZE + 1];
```

Comment:
Put these closer to their actual use. Top-of-function quit being a rule long ago. (Yes, a lot of the code in GPSBabel is really old...)